### PR TITLE
Edit songs

### PR DESCRIPTION
**Pull request by alex-schechter**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
+import { SongDetailsComponent } from './song/song-details/song-details.component';
+import { SongCreateComponent } from './song/song-create/song-create.component';
 
-const routes: Routes = [];
+const routes: Routes = [
+  {path: '', component: SongListComponent},
+  {path: 'create', component: SongCreateComponent},
+  {path: 'details/:id', component: SongDetailsComponent},
+  {path: 'edit/:id', component: SongCreateComponent},
+];
```
Comment (QuestionGuy) [inline src/app/app-routing.module.ts]: Does that mean that localhost:4200/create will point me to song create component?
Why not localhost:4200/song_create or something like that?

File: src/app/song/songs.service.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export class SongService{
                 private notificationService:NotificationPopupService){}
 
     getSongs(){
-        this.Http.get<{message: string; songs: any}>('http://localhost:3000/api/getSongs')
+        this.Http.get<{message: string; songs: any}>('http://localhost:3000/api/songs')
```
Comment (QuestionGuy) [inline src/app/song/songs.service.ts]: Put the base endpoint in some global variable in some consts file so if someone wants to run it on another port they won't have to change every single URL you've sent a request to.

File: src/app/song/songs.service.ts [hunk 3]
```diff
@@ -65,14 +69,34 @@ export class SongService{
...
+                id: id, 
+                name: name, 
+                genre: genre, 
+                song_path: song_path, 
+                image_path: image_path, 
+                release_date: release_date,
+                artists: artists, 
+                num_of_times_liked: num_of_times_liked
+            };
+            this.Http.put("http://localhost:3000/api/songs/" + id, song).subscribe(res => {
```
Comment (QuestionGuy) [inline src/app/song/songs.service.ts]: Same as above, do something like endpoint="http://localhost:3000"
and then here:
Http.put(endpoint + "/api/songs")

File: src/app/header/header.component.css [hunk 4]
```diff
@@ -0,0 +1,9 @@
+ul{
+    list-style: none;
+    padding: 0;
+    margin: 0;
+}
+
+.rightSide{
+    flex: 1, 1, auto;
```
[the remark this answers is not in the document]
Comment (alex-schechter) [reply src/app/header/header.component.css]: should b something else

File: src/app/header/header.component.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<mat-toolbar color="blue">
```
[the remark this answers is not in the document]
Comment (alex-schechter) [reply src/app/header/header.component.html]: no

**Pull request merged**